### ISSUE TITLE
swap() method for Quantity() / rebase() preparation

### ISSUE THREAD

**Issue by alextanski** [Quantipy/quantipy#375]
I will add a swap() feature to the Quantity() class, that makes it possible to change an axis on-the-fly without the need to actually have the related Link inside the stack at runtime. This can be used for a rebase() method that will exist alongside the regular .normalize() but offers the flexibility to define  base rows / columns or a full distribution to base the pct calculation on.

**Comment by alextanski** [comment]
```
l1 = stack[name_data]['no_filter']['q5_1']['q8']
q = qp.Quantity(l1)
print q.count()

q.swap(var='q5_3', axis='y', inplace=True)
print q.count()

q.swap(var='q5_3', axis='x', inplace=True)
print q.count()
```

```
Question           q8                                   
Values            All    1    2    3    4     5   96  98
Question Values                                         
q5_1     All     2367  949  216  595  970  1235  283  49
         4         28    9    2    6   18    16    1   0
         3       1014  418   95  260  393   517  138  20
         5        644  266   53  169  290   358   65  10
         2        227   94   27   52   88   120   21   3
         1         93   26   15   23   38    45    7   3
         97        28   13    1   10   11     9    5   1
         98       333  123   23   75  132   170   46  12
Question         q5_3                                      
Values            All    1     2     3   4     5   97    98
Question Values                                            
q5_1     All     8255  612  1245  2702  66  1710  309  1611
         4        124   12    22    27  26    23    6     8
         3       2598  147   315  1580  20   328   46   162
         5       2472  136   326   631  12  1108   51   208
         2        700   69   424    92   1    71   10    33
         1        363  193    49    63   0    32    6    20
         97       200    7     8    16   1    13  133    22
         98      1798   48   101   293   6   135   57  1158
Question         q5_3                                      
Values            All    1     2     3   4     5   97    98
Question Values                                            
q5_3     All     8255  612  1245  2702  66  1710  309  1611
         1        612  612     0     0   0     0    0     0
         2       1245    0  1245     0   0     0    0     0
         3       2702    0     0  2702   0     0    0     0
         4         66    0     0     0  66     0    0     0
         5       1710    0     0     0   0  1710    0     0
         97       309    0     0     0   0     0  309     0
         98      1611    0     0     0   0     0    0  1611
```

**Comment by alextanski** [comment]
```
l1 = stack[name_data]['no_filter']['q5_1']['q8']
q = qp.Quantity(l1)
print q.count(margin=False)
print q.rebase(reference='q5_2')
```

```
Values            1   2    3    4    5    96  98
Question Values                                 
q5_1     1        26  15   23   38   45    7   3
         2        94  27   52   88  120   21   3
         3       418  95  260  393  517  138  20
         4         9   2    6   18   16    1   0
Question          q8                            
Values            1   2    3    4    5    96  98
Question Values                                 
q5_2     98       80  16   43   86   95   31   7
         97        8   1    5    6    6    3   1
         5       188  42  127  208  263   53   7
         4         9   2    9   12   13    3   0
         3       417  89  275  412  535  123  25
         2       186  36  107  175  236   55   6
         1        61  30   29   71   87   15   3
Question                 q8                                                                
Values                   1           2          3           4           5           96   98
Question Values                                                                            
q5_1     1        42.622951   50.000000  79.310345   53.521127   51.724138   46.666667  100
         2        50.537634   75.000000  48.598131   50.285714   50.847458   38.181818   50
         3       100.239808  106.741573  94.545455   95.388350   96.635514  112.195122   80
         4       100.000000  100.000000  66.666667  150.000000  123.076923   33.333333  NaN
```

**Comment by jamesrkg** [comment]
Resolved by #388.
